cmsplugin_cascade/plugin_base.py

```diff
@@ -294,15 +294,28 @@ def save_model(self, request, new_obj, form, change):
                     new_obj.glossary[key] = old_obj.glossary[key]
         super(CascadePluginBase, self).save_model(request, new_obj, form, change)
 
-    def get_parent_instance(self, request=None):
+    def get_parent_instance(self, request=None, obj=None):
         """
         Get the parent model instance corresponding to this plugin. When adding a new plugin, the
         parent might not be available. Therefore as fallback, pass in the request object.
         """
         try:
-            parent_id = self.parent.id
+            parent_id = obj.parent_id
         except AttributeError:
-            parent_id = request.GET.get('plugin_parent') if request else None
+            try:
+                parent_id = self.parent.id
```

Comment:
sincerely, I believe that in CMS-3.4 a plugin class does not contain any reference to the parent anymore.
This was an inconsistent approach anyway, so I'm happy if it will go away.